code/datums/components/shrink.dm

```diff
@@ -0,0 +1,42 @@
+/datum/component/shrink
+	var/olddens
+	var/oldopac
+	dupe_mode = COMPONENT_DUPE_HIGHLANDER
+
+/datum/component/shrink/Initialize(shrink_time)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	var/atom/parent_atom = parent
+	parent_atom.transform = parent_atom.transform.Scale(0.5,0.5)
```

Comment:
Don't like this from the one simple fact that we can't trust players to NOT roleplay as smol beans

Reply:
It is a temporary effect, so it seems ok